File: be/src/sales-orders/dto/void-sales-order.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MinLength } from 'class-validator';

export class VoidSalesOrderDto {
  @ApiProperty({ description: 'Reason for voiding the transaction' })
  @IsString()
  @IsNotEmpty()
  reason: string;

  @ApiProperty({ description: 'Authorizer user ID (must be admin or supervisor)' })
  @IsString()
  @IsNotEmpty()
  authorizer_user_id: string;

  @ApiProperty({ description: 'Authorizer device PIN' })
  @IsString()
  @IsNotEmpty()
  @MinLength(4)
  authorizer_pin: string;
}

File: kiosk/lib/data/backend_api/schemas/void_sales_order_dto.dart
class VoidSalesOrderDto {
  const VoidSalesOrderDto({
    required this.reason,
    required this.authorizerUserId,
    required this.authorizerPin,
  });

  final String reason;
  final String authorizerUserId;
  final String authorizerPin;

  Map<String, dynamic> toJson() => {
    'reason': reason,
    'authorizer_user_id': authorizerUserId,
    'authorizer_pin': authorizerPin,
  };
}
